src/System.Runtime.InteropServices.RuntimeInformation/src/RuntimeInformation.Windows.cs

```diff
@@ -1,13 +1,77 @@
 ﻿// Copyright (c) Microsoft. All rights reserved.
 // Licensed under the MIT license. See LICENSE file in the project root for full license information.
 
+using System.Diagnostics;
+
 namespace System.Runtime.InteropServices
 {
-    public static class RuntimeInformation
+    public static partial class RuntimeInformation
     {
+        private static string s_osDebugName = null;
+        private static Architecture? s_osArch = null;
+
         public static bool IsOSPlatform(OSPlatform osPlatform)
         {
             return OSPlatform.Windows == osPlatform;
         }
+
+        public static string GetOSDebugName()
+        {
+            if (null == s_osDebugName)
+            {
+                string type = Interop.mincore.IsWindowsServer() ? "Server" : "Client";
+                string version = "Unknown";
+
+                if (Interop.mincore.IsWindows10OrGreater())
+                    version = "10";
```

Comment:
LOL... yes people could game the system in many ways I'm sure. I actually don't know how VerifyVersionInfo behaves myself but I would assume it doesn't lie. I think the windows folks would be very unhappy with us if we provided an API that figured out the version in this way and handed it out. I believe there primary goal is to make sure people properly future proof their code which is why they provide the IsWindowsXYZOrGreater macros. I don't think they have an API for getting a information string like we are looking for. 

With that said I agree with you that I don't want to have to rev this every time the OS revs. I wonder if there is some other API we can call in windows to get something useful here.